api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java

```diff
@@ -440,464 +539,1036 @@ public void setUp() {
+                workspace.getNamespace(), workspace.getId(), cohortReviewId, participantId, request)
+            .getBody();
+
+    assertCreatedParticipantCohortAnnotation(response, request);
+  }
+
+  @ParameterizedTest(
+      name = "createParticipantCohortAnnotationAllowedAccessLevel WorkspaceAccessLevel={0}")
+  @EnumSource(
+      value = WorkspaceAccessLevel.class,
+      names = {"OWNER", "WRITER"})
+  public void createParticipantCohortAnnotationAllowedAccessLevel(
+      WorkspaceAccessLevel workspaceAccessLevel) {
+    deleteExistingParticipantCohortAnnotations();
+    // change access, call and check
+    stubWorkspaceAccessLevel(workspace, workspaceAccessLevel);
+
+    Long cohortReviewId = cohortReview.getCohortReviewId();
+    Long participantId = participantCohortStatus1.getParticipantKey().getParticipantId();
+    // use stringAnnotationType
+    ParticipantCohortAnnotation request =
+        buildValidParticipantCohortAnnotationForType(
+            cohortReviewId, participantId, AnnotationType.STRING);
+    ParticipantCohortAnnotation response =
+        cohortReviewController
+            .createParticipantCohortAnnotation(
+                workspace.getNamespace(),
+                workspace.getId(),
+                cohortReview.getCohortReviewId(),
+                participantId,
+                request)
+            .getBody();
+
+    assertCreatedParticipantCohortAnnotation(response, request);
+  }
+
+  @ParameterizedTest(
+      name = "createParticipantCohortAnnotationForbiddenAccessLevels WorkspaceAccessLevel={0}")
+  @EnumSource(
+      value = WorkspaceAccessLevel.class,
+      names = {"READER", "NO_ACCESS"})
+  public void createParticipantCohortAnnotationForbiddenAccessLevels(
+      WorkspaceAccessLevel workspaceAccessLevel) {
+    // change access, call and check
+    stubWorkspaceAccessLevel(workspace, workspaceAccessLevel);
+
+    Long cohortReviewId = cohortReview.getCohortReviewId();
+    Long participantId = participantCohortStatus1.getParticipantKey().getParticipantId();
+    // use stringAnnotationType
+    ParticipantCohortAnnotation request =
+        buildValidParticipantCohortAnnotationForType(
+            cohortReviewId, participantId, AnnotationType.STRING);
+    Throwable exception =
+        assertThrows(
+            ForbiddenException.class,
+            () ->
+                cohortReviewController.createParticipantCohortAnnotation(
+                    workspace.getNamespace(),
+                    workspace.getId(),
+                    cohortReview.getCohortReviewId(),
+                    participantId,
+                    request));
+    assertForbiddenException(exception);
+  }
+
+  ////////// updateParticipantCohortAnnotation  //////////
+  @Test
+  public void updateParticipantCohortAnnotationNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.WRITER);
+    Long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```

Comment:
What is this sorcery? 
```suggestion
    Long wrongReviewId = -1L;
```

Reply:
done